### PR TITLE
fix: remove reconciliation of peerauth for PERMISSIVE mode

### PR DESCRIPTION
**Pull request by aslakknutsen**
Encryption between all services is a requirement, so no service should support PERMISSIVE
mode. Non mesh members should rely on the Public service names as provided
by Routing to talk simple tls to the mesh and the mesh is mtls internally.

> [!NOTE]
> Keeping peer auth code around for the time being just in case this
> assumption of PERMISSIVE mode not being needed falls short. The on boarding
> of more components will prove the thesis.

### CODE REVIEW

**Comment by bartoszmajsak** [comment]
Would be good if you expand in commit message/PR description why are you removing this step from reconcile and replacing it with what instead.

I also wonder if we shouldn't remove the func itself too. Where is it used now?

**Comment by aslakknutsen** [comment]
Updated commit/description

File: controllers/authorization/authorization_controller_test.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ var _ = Describe("Checking Authorization Resource Creation", test.EnvTest(), fun
 	// [1] https://book.kubebuilder.io/reference/envtest#testing-considerations
 	It("should have ownerReference on all created auth resources", func(ctx context.Context) {
 		Eventually(func() error {
-			createdPeerAuth := &istiosecurityv1beta1.PeerAuthentication{}
+			createdPeerAuth := &istiosecurityv1beta1.AuthorizationPolicy{}
```
Comment (bartoszmajsak) [inline controllers/authorization/authorization_controller_test.go]: Wrong variable name :)

File: controllers/authorization/authorization_controller_test.go [hunk 2]
```diff
@@ -140,7 +140,7 @@ var _ = Describe("Checking Authorization Resource Creation", test.EnvTest(), fun
 		}, test.DefaultTimeout, test.DefaultPolling).Should(Succeed())
 	})
 
-	It("should create a PeerAuthentication when a Component is created", func(ctx context.Context) {
+	XIt("should create a PeerAuthentication when a Component is created", func(ctx context.Context) {
```
Comment (bartoszmajsak) [inline controllers/authorization/authorization_controller_test.go]: x

I think we should just delete this code too?

File: controllers/authorization/authorization_reconcile_peerauthentication.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+//nolint:unused //reason no permissive mode required
 func (r *PlatformAuthorizationController) reconcilePeerAuthentication(ctx context.Context, target *unstructured.Unstructured) error {
```
Comment (bartoszmajsak) [inline controllers/authorization/authorization_reconcile_peerauthentication.go]: Why not just delete it?

File: controllers/authorization/authorization_reconcile_peerauthentication.go [hunk 4]
```diff
@@ -65,6 +66,7 @@ func (r *PlatformAuthorizationController) reconcilePeerAuthentication(ctx contex
 	return nil
 }
 
+//nolint:unused //reason no permissive mode required
```
Comment (bartoszmajsak) [inline controllers/authorization/authorization_reconcile_peerauthentication.go]: Same as above.

**Review by bartoszmajsak — CHANGES_REQUESTED** [review]

**Comment by aslakknutsen** [comment]
Season 9 Ok GIF by Curb Your Enthusiasm

**Review by bartoszmajsak — APPROVED** [review]
LGTM

**Pull request merged**